openshift/e2e-tests-openshift.sh

```diff
@@ -96,201 +63,25 @@ run_e2e_tests(){
-  sed -i -e "s|registry.svc.ci.openshift.org/openshift/knative-.*:knative-serving-controller|registry.svc.ci.openshift.org/openshift/knative-nightly:knative-serving-controller|g"         ${CATALOG_SOURCE}
-  sed -i -e "s|registry.svc.ci.openshift.org/openshift/knative-.*:kourier|registry.svc.ci.openshift.org/openshift/knative-nightly:kourier|g"                                               ${CATALOG_SOURCE}
-
-  envsubst < $CATALOG_SOURCE | oc apply -n $OLM_NAMESPACE -f -
-
-  # Replace kourier's image with the latest ones from third_party/kourier-latest
-  KOURIER_CONTROL=$(grep -w "gcr.io/knative-nightly/knative.dev/net-kourier/cmd/kourier" third_party/kourier-latest/kourier.yaml  | awk '{print $NF}')
-  KOURIER_GATEWAY=$(grep -w "docker.io/maistra/proxyv2-ubi8" third_party/kourier-latest/kourier.yaml  | awk '{print $NF}')
-
-  sed -i -e "s|docker.io/maistra/proxyv2-ubi8:.*|${KOURIER_GATEWAY}|g"                                        ${CATALOG_SOURCE}
-  sed -i -e "s|registry.svc.ci.openshift.org/openshift/knative-.*:kourier|${KOURIER_CONTROL}|g"               ${CATALOG_SOURCE}
-
-  # release-next branch keeps updating the latest manifest in knative-serving-ci.yaml for serving resources.
-  # see: https://github.com/openshift/knative-serving/blob/release-next/openshift/release/knative-serving-ci.yaml
-  # So mount the manifest and use it by KO_DATA_PATH env value.
-  patch -u ${CATALOG_SOURCE} < openshift/olm/config_map.patch
-
-  oc apply -n $OLM_NAMESPACE -f ${CATALOG_SOURCE}
-  timeout 900 '[[ $(oc get pods -n $OLM_NAMESPACE | grep -c serverless) -eq 0 ]]' || return 1
-  wait_until_pods_running $OLM_NAMESPACE
-
-  # Deploy Serverless Operator
-  deploy_serverless_operator
-
-  # Wait for the CRD to appear
-  timeout 900 '[[ $(oc get crd | grep -c knativeservings) -eq 0 ]]' || return 1
-
-  # Install Knative Serving
-  cat <<-EOF | oc apply -f -
-apiVersion: operator.knative.dev/v1alpha1
-kind: KnativeServing
-metadata:
-  name: knative-serving
-  namespace: ${SERVING_NAMESPACE}
-EOF
-
-  # Wait for 4 pods to appear first
-  timeout 900 '[[ $(oc get pods -n $SERVING_NAMESPACE --no-headers | wc -l) -lt 4 ]]' || return 1
-  wait_until_pods_running $SERVING_NAMESPACE || return 1
-
-  wait_until_service_has_external_ip $SERVING_INGRESS_NAMESPACE kourier || fail_test "Ingress has no external IP"
-  wait_until_hostname_resolves "$(kubectl get svc -n $SERVING_INGRESS_NAMESPACE kourier -o jsonpath='{.status.loadBalancer.ingress[0].hostname}')"
-
-  header "Knative Serving installed successfully"
+install_serverless_operator_branch() {
+  local branch={$1:-"master"}
+  local clone_path="/tmp/serverless-operator"
+  header "Installing Serverless Operator from branch: $branch"
+  rm -rf $clone_path
+  git clone --branch $branch https://github.com/openshift/knative-eventing.git $clone_path
+  pushd $clone_path
+  make install || return 1
+  header "Serverless Operator installed successfully"
   popd
```

Comment:
```
./openshift/e2e-tests-openshift.sh: line 75: popd: directory stack empty
```

Reply:
There's also error cloning repo.
```
git clone --branch '{master:-master}' https://github.com/openshift/knative-eventing.git /tmp/serverless-operator
Cloning into '/tmp/serverless-operator'...
fatal: Remote branch {master:-master} not found in upstream origin
Unexpected end of command stream
+ pushd /tmp/serverless-operator
./openshift/e2e-tests-openshift.sh: line 72: pushd: /tmp/serverless-operator: No such file or directory
+ make install 
```

Reply:
yup, the clone didnt happen because of incorrect URL and the dir doesnt exists, fixed it, wondering why it closed the PR :thinking: